pkg/queryexpr/sql_expr_test.go

```diff
@@ -0,0 +1,97 @@
+package queryexpr_test
+
+import (
+	"testing"
+
+	"github.com/goto/compass/pkg/queryexpr"
+)
+
+func TestSQLExpr_String(t *testing.T) {
+	tests := []struct {
+		expr queryexpr.SQLExpr
+		want string
+	}{
+		{
+			expr: queryexpr.SQLExpr("test"),
+			want: "test",
+		},
+		{
+			expr: queryexpr.SQLExpr("bool_identifier == !(findLast([1, 2, 3, 4], # > 2) == 4)"),
+			want: "bool_identifier == !(findLast([1, 2, 3, 4], # > 2) == 4)",
+		},
+	}
+	for i, tt := range tests {
+		t.Run("test-case-"+string(rune(i)), func(t *testing.T) {
+			if got := tt.expr.String(); got != tt.want {
+				t.Errorf("String() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestSQLExpr_ToQuery(t *testing.T) {
+	tests := []struct {
+		name    string
+		expr    queryexpr.SQLExpr
+		want    string
+		wantErr bool
+	}{
+		{
+			name:    "less than condition with single quote",
+			expr:    queryexpr.SQLExpr(`updated_at < '2024-04-05 23:59:59'`),
+			want:    `(updated_at < '2024-04-05 23:59:59')`,
+			wantErr: false,
+		},
+		{
+			name:    "greater than condition with double quote",
+			expr:    queryexpr.SQLExpr(`updated_at > "2024-04-05 23:59:59"`),
```

Comment:
can we support this syntax as well mas?
```
refreshed_at <= (now() - duration('7d'))
```

should translate to for example
```
refreshed_at <= '2024-08-14 15:54:00'
```

Reply:
Duration in` expr-lang` itself only capable to handle in hours,
so if you mean is `refreshed_at <= (now() - duration('168h'))` then the answer is yes mas

Reply:
oh ok, no problem, the idea is to make it easier from client perspective instead of calculating it manually, can you add new test case then?

Reply:
Added as new test case